crates/bevy_ecs/src/query/state.rs

```diff
@@ -1979,12 +1988,12 @@ mod tests {
     fn can_transmute_filtered_entity() {
         let mut world = World::new();
         let entity = world.spawn((A(0), B(1))).id();
-        let query =
-            QueryState::<(Entity, &A, &B)>::new(&mut world).transmute::<FilteredEntityRef>(&world);
+        let query = QueryState::<(Entity, &A, &B)>::new(&mut world)
+            .transmute::<(Entity, FilteredEntityRef)>(&world);
```

Comment:
How come the tests needs an extra 'Entity' in the signature?

Reply:
That was just to make the tests start failing before I made the other changes :).